components/pd_client/src/client.rs

```diff
@@ -398,25 +402,23 @@ impl PdClient for RpcClient {
             tx.unbounded_send(req)
                 .unwrap_or_else(|e| panic!("send request to unbounded channel failed {:?}", e));
             inner.hb_sender = Either::Right(tx);
-            Box::new(
-                sender
+            Box::pin(async move {
+                let result = sender
+                    .sink_compat()
                     .sink_map_err(Error::Grpc)
-                    .send_all(rx.then(|r| match r {
-                        Ok(r) => Ok((r, WriteFlags::default())),
-                        Err(()) => Err(Error::Other(box_err!("failed to recv heartbeat"))),
-                    }))
-                    .then(|result| match result {
-                        Ok((mut sender, _)) => {
-                            info!("cancel region heartbeat sender");
-                            sender.get_mut().cancel();
```

Comment:
Why remove this?

Reply:
I have added it back, but can you explain why cannot rely on `drop` to cancel a call

Reply:
/cc @overvenus. I believe it's added explicitly to fix a bug.